### ISSUE TITLE
hashlib constructors unexpected-argument false positive for `usedforsecurity`

### ISSUE THREAD

**Issue by OJFord** [PyCQA/pylint#6017]
### Bug description

Python 3.9 added the `usedforsecurity` keyword-only argument to all `hashlib` constructors

### Command used

```shell
pylint
```

### Pylint output

```shell
E1123: Unexpected keyword argument 'usedforsecurity' in constructor call (unexpected-keyword-arg)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.0.0-a4
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. Could you try with the latest pylint <3 ? 3.0.0 alpha 4 is based on an old version of pylint like 2.10 with breaking change on top of it.

**Comment by OJFord** [comment]
Thanks for the reply - I think it might actually be that python3.8 was (unexpectedly) picked up, in which case the error is correct. Will close once I've confirmed that.

**Comment by LuminairPrime** [comment]
`Unexpected keyword argument 'usedforsecurity' in constructor call`

```
from hashlib import blake2b
blake2bhash = blake2b(digest_size=10, usedforsecurity=False)
```

```
pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.10.1 (tags/v3.10.1:2cd268a, Dec  6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
pylint uses astroid to create the abstract syntax tree. It's possible there's an issue in astroid not being able to create the proper AST for the python 3.9 code.

**Comment by OJFord** [comment]
Thanks! (Aside: didn't realise you could close cross-repo like that now, nice!)
